### PR TITLE
chore: migrate npx react-native init

### PR DESCRIPTION
**Pull request by coado**
## Summary

Migrating deprecated `npx react-native init` command to `npx @react-native-community/cli init` as mentioned [here](https://github.com/software-mansion/react-native-reanimated/pull/6533#pullrequestreview-2321459011)

### CODE REVIEW

**Review by tomekzaw — APPROVED** [review]

**Comment by tomekzaw** [comment]
Looks good but CIs are failing

**Comment by tjzel** [comment]
Let me fix Yarn & installing from commit management in https://github.com/software-mansion/react-native-reanimated/pull/6635 prior to merging it

File: .github/workflows/V8-reanimated-build-check-nightly.yml [hunk 1]
```diff
@@ -29,13 +29,13 @@ jobs:
           distribution: 'zulu'
           java-version: '17'
       - name: Create React Native app
-        run: npx react-native init app
```
Comment (tjzel) [inline .github/workflows/V8-reanimated-build-check-nightly.yml:32]: `npx react-native init`/`rncli init` can be unstable at times. Installing it firstly with yarn, then doing `yarn rncli init` improves stability for some reasons. Please adjust to that way of app initialization here.

**Review by tjzel — CHANGES_REQUESTED** [review]

**Comment by tjzel** [comment]
I'm doing an aggressive takeover 😈

**Review by tjzel — APPROVED** [review]
Still doesn't work but it doesn't work in a better way now 🥳

**Pull request merged**